runtime/tests/checker/reference_test.go

```diff
@@ -830,17 +827,15 @@ func TestCheckInvalidReferenceResourceLoss(t *testing.T) {
       fun test() {
           let rs <- [<-create R()]
           let ref = &rs as &[R]
-          ref[0]
+          ref[0]  // This result in a reference, so no resource loss
```

Comment:
Do we keep a test for what was originally tested, indexing returning a resources and it not being dropped implicitly?

Reply:
Not really sure. All the tests in this file are for reference testing. There is a separate set of tests for resources in the `checker/reference_test.go` file. There are quite a lot of tests there, so hard find whether the exact test is already available. So added a new one anyway https://github.com/onflow/cadence/pull/2573/commits/498c63a860a8d24c8f5765689416d33705499573.